### ISSUE TITLE
Sea Otter Deadlock with seemingly no DB errors in Log

### ISSUE THREAD

**Issue by D3nnis3n** [StrangeLoopGames/EcoIssues#21580]
[log_210425062126.log](https://github.com/StrangeLoopGames/EcoIssues/files/6372650/log_210425062126.log)

As usual: Hangs, kicks people out, cant be joined, still uses 20% CPU and 10 GB of RAM and is up, but can't be used.

I made a memory dump, maybe its useful:
https://drive.google.com/file/d/17HfhI0RTqSJLtUbl8qx7tXMS-8Ypq6Ja/view?usp=sharing

**Comment by D3nnis3n** [comment]
[log_210425091020.log](https://github.com/StrangeLoopGames/EcoIssues/files/6373394/log_210425091020.log)

Leading to this, locking all servers:
![image](https://user-images.githubusercontent.com/25908592/116012372-5df35f00-a62a-11eb-8b00-1bf1788373c5.png)

Profiler not wanting to attach:
![image](https://user-images.githubusercontent.com/25908592/116012378-66e43080-a62a-11eb-8b87-2ba38d4706f7.png)

Memory Dump (50 GB), then killed the server and shutdown for now as it impedes all servers:

**Comment by mirasrael** [comment]
Investigated the issue. Seems like it happens because LiteDB goes mad: https://github.com/mbdavid/LiteDB/pull/1994
Created issue for LiteDB.
Alos I realized that Timeout issue happens not because of long transactions, but because of Checkpoints (which flushes changes to disk). In 5.0.10 checkpoints behavior was improved (I submitted PR which was accepted and released), so probably it makes sense to update to 5.0.10 and see if things improved.
